scripts/bluetooth/mesh/mesh_dfu_metadata.py

```diff
@@ -0,0 +1,425 @@
+            comp (dict): A Composition Data dictionary class object to be filled in
+
+        Returns:
+            TBA
+        """
+        for addr in self.addrs:
+            self.__read_comp_data(addr, kconfigs, comp)
+
+
+def parse_kconfig(config_path):
+    """
+    Extracts content of '.config' file into a dictionary
+
+    Parameters:
+        config_path (str): Path to '.config' file used by a firmware
+    Returns:
+        dict: A dictionary where key is Kconfig option name and value is the value of the corresponding Kconfig option.
+    """
+    with open(config_path, 'r') as config:
+        configs = {}
+        for line in config:
+            if not line.startswith("CONFIG_"):
+                continue
+            kconfig, value = line.split("=")
+            configs[kconfig] = value.split("\n")[0]
+        return configs
+
+def version_parse(cfg):
+    if 'CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION' not in cfg.keys():
+        print("The CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION Kconfig option not found in .config file")
+        return None
+
+    try:
+        clean_str = cfg['CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION'].replace("+", ".").replace("\"", "")
+        version_list = [int(s) for s in clean_str.split(".") if s.isdigit()]
+        return {
+            "major": version_list[0],
+            "minor": version_list[1],
+            "revision": version_list[2],
+            "build_number": version_list[3],
+        }
+    except Exception:
+        print("Unable to parse CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION Kconfig option")
+
+if __name__ == "__main__":
+    if len(sys.argv) < 2:
+        print("Missing build folder argument")
+        sys.exit(1)
+
+    bin_path = sys.argv[1]
+    file_name_in_zip = 'mesh_metadata.json'
+    zip_path = os.path.abspath(bin_path + '/dfu_application.zip')
+    file_name = 'dfu_application.zip_mesh_metadata.json'
+    metadata_path = os.path.abspath(bin_path + '/' + file_name)
+    elf_path = os.path.abspath(bin_path + '/zephyr.elf')
+    config_path = os.path.abspath(bin_path + '/.config')
+
+
+    zip = ZipFile(zip_path, "a")
+    if file_name_in_zip in zip.namelist():
+        # Mesh metadata already present in zip file
+        sys.exit(0)
+
+    comp = Comp0()
+    kconfigs = parse_kconfig(config_path)
+    elf_parser = CompElfParser(elf_path)
+
+    comp.features_add(kconfigs)
+    elf_parser.extract_comp_data(kconfigs, comp)
+
+    j_dict = {
+        "sign_version": version_parse(kconfigs),
+        "composition_hash": str(hex(comp.hash_generate())),
+        "composition_data": comp.dict_generate(),
+    }
+
+    with open(metadata_path, "w") as outfile:
+        outfile.write(json.dumps(j_dict, indent=4))
+    zip.write(metadata_path, file_name_in_zip)
+    zip.close()
+
+    print("Bluetooth Mesh Composition metadata generated:")
+    print("\tMetadata hash: {}".format(j_dict["composition_hash"]))
```

Comment:
You mean the composition+version+size etc as a string?

Reply:
Yes, which is the output of `metadata encode` shell command: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/zephyr/connectivity/bluetooth/api/mesh/shell.html#mesh-models-dfu-metadata-encode-major-minor-rev-buildnum-size-coretype-hash-elems-userdata

Reply:
After I tried this I see that either remove this line as it is not needed, or better do as Pavel suggest and print the full encoded metadata string. If it is needed in the file it is optional as the file be read by some machine/script and that can create the encoded string. But I do not think it hurt either.

Reply:
I agree, the bytstring should be the entire string you need to include in the message. I also think it should be included in the json, this will make it easier for customers wanting to make their own initiators for use with our DFU solution (on platforms where the Mobile App lib is not availble) and is easy to do.